gulpfile.js

```diff
@@ -67,12 +67,50 @@ function handlebarTemplates () {
  */
 function handlebarRuntime () {
   return gulp.src('node_modules/handlebars/dist/cjs/**/*')
-    .pipe(gulp.dest('source/common/assets/handlebars/'))
+    .pipe(gulp.dest('dist/common/assets/handlebars/'))
+}
+
+/**
+ * Copies all static assets to the "dist" folder
+ */
+function copyAssets () {
+  return gulp.src([
+    'source/**/assets/**/*'
+  ])
+    .pipe(sourcemaps.init())
+    .pipe(sourcemaps.write())
+    .pipe(gulp.dest('dist'))
+}
+
+/**
+ * Copies all vendor-specific assets to the "dist" folder
+ */
+function copyVendorAssets () {
+  return gulp.src([
+    'node_modules/katex/dist/katex.min.css',
```

Comment:
this whole stuff should be coordinated with #938 because I already see a clash of modules here

Reply:
Yeah...I'm not yet convinced that the "electron-webpack" project is the right way to go...but I don't have any experience with this and was just judging from the activity of the project. Also I don't like that it relies on so many assumptions where the code needs to lie etc. What's your opinion?

Reply:
Yes, this change will clash with the webpack change.

@tobiasdiez, I'm not sure what you mean by assumptions about code structure. Webpack can locate packages installed by npm/yarn automatically and for me that means less assumptions than having explicit references to node_modules in the code, because that relies on relative path to something. We might look at different things, so just let me know. I'm happy to discuss this more in depth separately.

As for the electron-webpack change, I have an updated version locally that I haven't pushed yet. The remaining broken thing is the theme switching. I'm thinking about opening a precursor PR for restructuring themes to make that shift easier and not further inflate the scope of the electron-webpack change. What do you two think about phasing out Less and using CSS variables only? (I have not seen other Less specific features used, but quite possible I missed something.)

As for a longer term perspective, as @tobiasdiez mentioned, Electron Forge is probably the right way to head and I missed it when I just searched for webpack and Electron. However, Electron Forge should be used with webpack as well, so switching from electron-webpack should be fairly straight forward as the main migration cost comes from standardising import paths across the codebase.

Reply:
Thanks @Aigeruth for chiming in! I really like your PR #938 and realized to late that the typescript-related changes here solve similar problems as you did. So we should definitely coordinate how to proceed from here, otherwise we duplicate work for no reason.

Concerning the "electron-webpack" framework, I have the impression that it relies a lot on 'convention over configuration', i.e. certain files have to lie in a certain location since otherwise nothing works. The gulp-approach on the other hand is really flexible (at the cost of being more verbose). But I realize that this initial skepticism might come from my unfamiliarity with the webpack world.

I just created a new test project using `yarn create electron-app my-new-app --template=typescript-webpack` (which uses electron forge with typescript and webpack support). This looks very similar to what you have in #938, so I would suggest to convert your PR to use electron-forge, and I will wait with this PR here until yours is merged. If you want we can also work together on #938. What you do you think?

Reply:
Sounds good, I'm happy collaborate on this migration as it is a bigger piece than I originally anticipated. I'm keen on breaking #938 into multiple PRs because it is just too risky rolling a PR that modifies 200-300 files.

I have raised #1055 to outline my thinking on how we could gradually prepare for the Electron Forge migration and make it a series of smaller changes. This could enable easier collaboration as well. We could even create a project board for this. Please, let me know what you think.

I see what you mean. I generally prefer 'convention over configuration' because it means that the convention is more likely to be documented and potential contributors are more likely to be familiar with it than something that I could create with a configuring something from scratch, therefore it enable better/faster collaboration in my view. It also allows me to focus on the important bits and not on the configuration. I have this somewhat under the same category as linting where I happily trade my personal style/taste for a standard way for collaboration.

Reply:
Good points, and I think what you two are describing here is indeed something we should have a longer chat about, as I assume both the migration as well as a potential rollback would be super time-consuming, so we should discuss to make sure the right approach is taken — what would you say we meet for a talk on Jitsi this week?

Please just send me a mail so that we can take this off GitHub :) (hendrik@zettlr.com)